feat(linter): Add options for `typescript/no-meaningless-void-operator` rule.

connorshea opened the pull request:
This option was already implemented on the tsgolint side, this just adds the config to the oxlint side.

See https://github.com/oxc-project/tsgolint/blob/main/internal/rules/no_meaningless_void_operator/options.go

Part of #16978.

```md
## Configuration

This rule accepts a configuration object with the following properties:

### checkNever

type: `boolean`

default: `false`

Whether to check `void` applied to expressions of type `never`.
```

camc314 commented:
### Merge activity

* **Dec 17, 9:11 AM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 17, 9:11 AM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).
* **Dec 17, 9:16 AM UTC**: Merged by the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
feat(linter): Add options for `typescript/no-meaningless-void-operator` rule. (#16981)

This option was already implemented on the tsgolint side, this just adds the config to the oxlint side.

See https://github.com/oxc-project/tsgolint/blob/main/internal/rules/no_meaningless_void_operator/options.go

Part of #16978.

```md
## Configuration

This rule accepts a configuration object with the following properties:

### checkNever

type: `boolean`

default: `false`

Whether to check `void` applied to expressions of type `never`.
```
